File: take_home/pirates/js/pirateShip.js
const { randomCoordinates } = require('./utils')

class PirateShip {
    constructor() {
        const coordinates = randomCoordinates();
        this.x = coordinates.x;
        this.y = coordinates.y;
        this.cargo = 0;
        this.money = 0;
    }
    //BUG: Pirate not bound to board.  Can fly off the board.
    moveNorth() {
        this.x -= 1;
    }
    moveSouth() {
        this.x += 1;
    }
    moveWest() {
        this.y -= 1;
    }
    moveEast() {
        this.y += 1;
    }
    //NOTE: why cant a pirate ship plunder another pirate ship (in 2 player)? 
    plunder(cargoShip) {
        this.cargo += cargoShip.cargo;
        this.money += cargoShip.money;
    }
}

module.exports = PirateShip;

File: take_home/pirates/python/pirate_ship.py
import utils


class PirateShip:
    def __init__(self):
        self.x, self.y = utils.random_coordinates()
        self.cargo = 0
        self.money = 0

    def move_north(self):
        self.y -= 1

    def move_south(self):
        self.y += 1

    def move_east(self):
        self.x += 1

    def move_west(self):
        self.x -= 1

    def plunder(self, cargo_ship):
        self.cargo = cargo_ship.cargo
        self.money = cargo_ship.money
